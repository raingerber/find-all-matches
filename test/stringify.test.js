/* eslint-env jest */

const {stringifySelectors} = require('../__test__/index')

describe('stringifySelectors', () => {
  it('should stringify selector arrays without modifying other keys', () => {
    const input = {
      matches: [{
        cssText: '<placeholder>',
        selector: [
          ['div', ' '],
          ['div', '.class']
        ]
      }],
      children: [{
        matches: [{
          cssText: '<placeholder>',
          selector: [
            ['div >', ' div ']
          ]
        }],
        children: []
      }, {
        matches: [{
          cssText: '<placeholder>',
          mediaText: 'max-width: 888px',
          selector: [
            [' div > ', ' div div'],
            ['', 'div']
          ]
        }],
        children: []
      }]
    }
    const options = {
      formatSelector: (a, b) => {
        return [a.trim() ? `?${a.trim()}?` : a, b.toUpperCase()]
      }
    }
    const result = stringifySelectors(input, options)
    expect(result).toMatchSnapshot()
  })
})
