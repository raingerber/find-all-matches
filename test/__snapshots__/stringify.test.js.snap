// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stringifySelectors should stringify selector arrays and convert cssText strings to arrays 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "matches": Array [
        Object {
          "css": "div > div { color: blue; margin: 0px 0px 10px 20px; }",
          "selector": "?div >? DIV",
        },
      ],
    },
    Object {
      "children": Array [],
      "matches": Array [
        Object {
          "css": "div > div div, div {}",
          "mediaText": "max-width: 888px",
          "selector": "?div >? DIV DIV, DIV",
        },
      ],
    },
  ],
  "matches": Array [
    Object {
      "css": "div, div .class { padding: 40px; font-size: 20px; }",
      "selector": "?div?, ?div? .CLASS",
    },
  ],
}
`;
